job_executor/worker/steps/dataset_pseudonymizer.py

```diff
@@ -20,192 +22,131 @@ def _get_unit_types(
+    unique_identifiers = compute.unique(
+        identifiers_table[column_name]
+    ).to_pylist()
+
     identifier_to_pseudonym = pseudonym_service.pseudonymize(
-        list(unique_identifiers), unit_id_type, job_id
+        unique_identifiers, unit_id_type, job_id
     )
-    output_csv_path = (
-        input_csv_path.parent / f"{input_csv_path.stem}_pseudonymized.csv"
-    )
-    target_file = open(output_csv_path, "w", newline="", encoding="utf-8")
-    with open(input_csv_path, newline="", encoding="utf-8") as csv_file:
-        for line in csv_file:
-            row = line.strip().split(";")
-            line_number: int = row[0]
-            unit_id: str = row[1]
-            value: str = row[2]
-            start_date: str = row[3]
-            stop_date: str = row[4]
-            target_file.write(
-                ";".join(
-                    [
-                        str(line_number),
-                        str(identifier_to_pseudonym[unit_id]),
-                        value,
-                        start_date,
-                        stop_date,
-                    ]
-                )
-                + "\n"
-            )
-    target_file.close()
-    return output_csv_path
 
+    return [
+        identifier_to_pseudonym[identifier] for identifier in identifiers_list
+    ]
 
-def _pseudonymize_measure_only(
-    input_csv_path: Path, unit_id_type: str, job_id: str
-) -> Path:
-    unique_measure_values = set()
-    with open(input_csv_path, newline="", encoding="utf-8") as csv_file:
-        for line in csv_file:
-            value = line.strip().split(";")[2]
-            unique_measure_values.add(value)
-    value_to_pseudonym = pseudonym_service.pseudonymize(
-        list(unique_measure_values), unit_id_type, job_id
-    )
-    output_csv_path = (
-        input_csv_path.parent / f"{input_csv_path.stem}_pseudonymized.csv"
-    )
-    target_file = open(output_csv_path, "w", newline="", encoding="utf-8")
-    with open(input_csv_path, newline="", encoding="utf-8") as csv_file:
-        for line in csv_file:
-            row = line.strip().split(";")
-            line_number: int = row[0]
-            unit_id: str = row[1]
-            value: str = row[2]
-            start_date: str = row[3]
-            stop_date: str = row[4]
-            target_file.write(
-                ";".join(
-                    [
-                        str(line_number),
-                        unit_id,
-                        str(value_to_pseudonym[value]),
-                        start_date,
-                        stop_date,
-                    ]
-                )
-                + "\n"
-            )
-    target_file.close()
-    return output_csv_path
+
+def _pseudonymize_if_needed(
+    dataset: dataset.FileSystemDataset,
+    column_name: str,
+    unit_id_type: Optional[str],
+    job_id: str,
+) -> Optional[List[str]]:
+    """
+    Pseudonymizes a column if a valid unit ID type is provided. Returns None otherwise.
```

Comment:
It looks like the function returns None if `unit_id_type` is None, any other value (invalid unit ID type as well) will try to execute.

Reply:
Yes you are correct it would be better if it checked for correct unit_id_type and return None if incorrect

Reply:
Considering the code that generates the unit_id_type variable it can only be None or a valid unit_type 👍🏻 So there is no chance of an invalid unit_id_type at this point. This could be way clearer to readers of the code if we defined a UnitIdType enum in tools and used that as the type of the variable here ex.: ```Union[None, UnitIdType]```